addons/base_import/models/base_import.py

```diff
@@ -615,11 +637,24 @@ def _parse_import_data(self, data, import_fields, options):
                             except Exception as e:
                                 raise ValueError(_("Error Parsing Date [%s:L%d]: %s") % (name, num + 1, e))
 
-            elif field['type'] in ('float', 'monetary') and name in import_fields:
+            elif field['type'] in ('float', 'monetary'):
                 # Parse float, sometimes float values from file have currency symbol or () to denote a negative value
                 # We should be able to manage both case
-                index = import_fields.index(name)
                 self._parse_float_from_data(data, index, name, options)
+
+            elif field['type'] == 'binary' and field.get('attachment'):
+                for num, line in enumerate(data):
+                    if re.match("^(http|https)://", line[index], re.IGNORECASE):
+                        try:
+                            bin_file = requests.get(line[index])
+                            Image.open(io.StringIO(bin_file.content))
```

Comment:
StringIO is not going to work here, image data is binary, so this needs to be `io.BytesIO`.

Also I expect this attempts to validate that the URL is a valid image? IIRC Image.open is lazy so it's not going to actually do anything, you need to force PIL to load/check it.

And then there's a DOS risk with respect to slowloris or [decompression bomb](http://www.aerasec.de/security/advisories/decompression-bomb-vulnerability.html) attacks. You may want to check the Content-Lenght of the response, and afterwards the advertised size of the image (IIRC in web_editor we limit incoming image to 42 megapixel) (/cc @odony)